Signal/src/ViewControllers/MediaPageViewController.swift

```diff
@@ -0,0 +1,706 @@
+        // more fine grained control
+        navController.modalPresentationStyle = .custom
+        navController.navigationBar.barTintColor = UIColor.ows_materialBlue
+        navController.navigationBar.isTranslucent = false
+        navController.navigationBar.isOpaque = true
+
+        // We want to animate the tapped media from it's position in the previous VC
+        // to it's resting place in the center of this view controller.
+        //
+        // Rather than animating the actual media view in place, we animate the presentationView, which is a static
+        // image of the media content. Animating the actual media view is problematic for a couple reasons:
+        // 1. The media view ultimately lives in a zoomable scrollView. Getting both original positioning and the final positioning
+        //    correct, involves manipulating the zoomScale and position simultaneously, which results in non-linear movement,
+        //    especially noticeable on high resolution images.
+        // 2. For Video views, the AVPlayerLayer content does not scale with the presentation animation. So you instead get a full scale
+        //    video, wherein only the cropping is animated.
+        // Using a simple image view allows us to address both these problems relatively easily.
+        self.view.alpha = 0.0
+
+        self.pagerScrollView.isHidden = true
+        self.presentationView.isHidden = false
+        self.presentationView.layer.cornerRadius = OWSMessageCellCornerRadius
+
+        fromViewController.present(navController, animated: false) {
+
+            // 1. Fade in the entire view.
+            UIView.animate(withDuration: 0.1) {
+                self.replacingView?.alpha = 0.0
+                self.view.alpha = 1.0
+            }
+
+            self.presentationView.superview?.layoutIfNeeded()
+            self.applyFinalMediaViewConstraints()
+
+            // 2. Animate imageView from it's initial position, which should match where it was
+            // in the presenting view to it's final position, front and center in this view. This
+            // animation duration intentionally overlaps the previous
+            UIView.animate(withDuration: 0.2,
+                delay: 0.08,
+                options: .curveEaseOut,
+                animations: {
+
+                self.presentationView.layer.cornerRadius = 0
+                self.presentationView.superview?.layoutIfNeeded()
+
+                // We must lay out once *before* we centerMediaViewConstraints
+                // because it uses the imageView.frame to build the constraints
+                // that will center the imageView, and then once again *after*
+                // to ensure that the centered constraints are applied.
+                    //TODO
+//                [self centerMediaViewConstraints];
+//                [self.mediaView.superview layoutIfNeeded];
```

Comment:
?

Reply:
Removed - I can no longer reproduce the behavior that this code, which used to live in MessageDetailViewController used to fix.